.github/workflows/docs.yml

```diff
@@ -50,9 +52,19 @@ jobs:
         env: 
           RUSTDOCFLAGS: "--enable-index-page -Zunstable-options"
 
-      - name: deploy
+      - name: Deploy stable docs
+        if: ${{ github.event_name == 'release' }}
+        uses: peaceiris/actions-gh-pages@v3
+        with:
+          publish_dir: target/doc/
+          deploy_key: ${{ secrets.ACTIONS_DEPLOY_KEY }}
+          external_repository: gtk-rs/gtk-rs.github.io
+          destination_dir: docs/stable/
+      - name: Deploy unstable docs
         if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/master' }}
         uses: peaceiris/actions-gh-pages@v3
         with:
-          github_token: ${{ secrets.GITHUB_TOKEN }}
-          publish_dir: ./target/doc/
+          publish_dir: target/doc/
+          deploy_key: ${{ secrets.ACTIONS_DEPLOY_KEY }}
```

Comment:
deploy_key can't be secrets.GITHUB_TOKEN?

Reply:
We can't use GITHUB_TOKEN since the docs get pushed to a different repo and therefore it doesn't have the correct permissions. See: https://github.com/peaceiris/actions-gh-pages#%EF%B8%8F-set-ssh-private-key-deploy_key

Reply:
Yeah except I don't think we need that extra repository for docs anymore, gh-pages branch should be more than enough iirc. But I would wait for @GuillaumeGomez's opinion on this one

Reply:
I wasn't talking about the `docs` repo. The docs are pushed to the webpage repo.

Reply:
What would be different between using the `gh-pages` branch & a new repository here? iirc, it would work exactly the same way.

Reply:
it still leave the odd situation with `docs-src` and `docs` like we have right now: https://github.com/gtk-rs/gtk-rs.github.io/issues/105

Reply:
> What would be different between using the gh-pages branch & a new repository here?
There is no new repository, the docs are pushed directly to https://github.com/gtk-rs/gtk-rs.github.io

Reply:
Like said in a few places: I want to keep the documentation repository on its own because it's a mess, whatever the way we use to handle it.